src/Data/Name.hs

```diff
@@ -0,0 +1,41 @@
+{-# LANGUAGE UnliftedNewtypes #-}
+
+module Data.Name
+  ( Name (Name),
+  )
+where
+
+import Data.Kind (Type)
+import Data.Proxy (Proxy (Proxy))
+import GHC.OverloadedLabels (IsLabel (fromLabel))
+import GHC.TypeLits (KnownSymbol, Symbol, symbolVal)
+
+-- -------------------------------------------------------------------------------------------------
+
+-- | 'Name' is the type of spectacle names. Using the OverloadedLabels syntax should be used to
+-- construct 'Name's.
+--
+-- @
+-- >>> :set -XOverloadedLabels -XPartialTypeSignatures
+-- >>> :t #myVar :: 'Name' _
+-- >>> #myVar :: Name _ :: Name "myVar"
+-- @
+--
+-- @since 0.1.0.0
+data Name :: Symbol -> Type where
+  Name :: KnownSymbol s => Proxy s -> Name s
+
+-- | @since 0.1.0.0
+instance (KnownSymbol s, l ~ s) => IsLabel l (Name s) where
+  -- The @l ~ nm@ unification immediately solves @nm@ so that 'IsLabel' doesn't
+  -- leave it ambiguous with a wanted constraint.
+  fromLabel = Name Proxy
```

Comment:
Looks like you renamed `nm` to `s`, or vice versa, 'cause the comment and the code don't match.